File: src/main/java/nu/peg/slack/pt/di/factory/DevelopmentSlackApiFactory.java
package nu.peg.slack.pt.di.factory;

import nu.peg.slack.pt.api.slack.DevelopmentLoggingSlackApi;
import nu.peg.slack.pt.api.slack.SlackApi;

import org.glassfish.hk2.api.Factory;

public class DevelopmentSlackApiFactory implements Factory<SlackApi> {

    @Override
    public SlackApi provide() {
        return new DevelopmentLoggingSlackApi();
    }

    @Override
    public void dispose(SlackApi instance) {
    }
}

File: src/main/kotlin/nu/peg/slack/pt/di/factory/DevelopmentSlackApiFactory.kt
package nu.peg.slack.pt.di.factory

import nu.peg.slack.pt.api.slack.DevelopmentLoggingSlackApi
import nu.peg.slack.pt.api.slack.SlackApi

import org.glassfish.hk2.api.Factory

class DevelopmentSlackApiFactory : Factory<SlackApi> {

    fun provide(): SlackApi {
        return DevelopmentLoggingSlackApi()
    }

    fun dispose(instance: SlackApi) {}
}
